### PR TITLE
Us127024/refactor categories

### PR DESCRIPTION
**Pull request by ronan-f**
https://rally1.rallydev.com/#/29180338367d/custom/486232622040?detail=%2Fuserstory%2F521276679604

Categories is now used by both assignments and quizzes so just moving everything up a level.

### CODE REVIEW

File: src/hypermedia-constants.js [hunk 1]
```diff
@@ -582,12 +590,7 @@ export const Actions = {
 		updateFilesSubmissionLimit: 'update-files-submission-limit',
 		updateDefaultScoringRubric: 'update-default-scoring-rubric',
 		updateNotificationEmail: 'update-notification-email',
-		add: 'add',
-		categories: {
-			select: 'select',
-			deselect: 'deselect',
-			add: 'add'
-		}
+		add: 'add'
```
Comment (mostlyfabulous) [inline src/hypermedia-constants.js:593]: duplicate of: https://github.com/BrightspaceHypermediaComponents/siren-sdk/pull/432/files#diff-b5ac48af05ca446b8b131d08b7a37fece756d6656a71b1d10d3d22920cd0bfa9R546 ?

Comment (ronan-f) [reply src/hypermedia-constants.js]: Good catch. Fixed!

**Review by mostlyfabulous — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
   "author": "",
   "license": "Apache-2.0",
   "dependencies": {
-    "@brightspace-ui/core": "^1",
+    "@brightspace-ui/core": "^1.207.3",
```
Comment (dustxd) [inline package.json:18]: Do you still need this?

Comment (ronan-f) [reply package.json]: Good point. Let me undo.

**Review by dustxd — APPROVED** [review]

File: test/activities/CategoriesEntity.js [hunk 3]
```diff
@@ -138,7 +138,7 @@ describe('CategoriesEntity', () => {
 
 	describe('Saves', () => {
 		it('saves categoryId', async() => {
-			await fetchMock.patchOnce('https://afe99802-9130-4320-a770-8d138b941e74.assignments.api.proddev.d2l/6606/folders/13', editableEntity);
+			await fetchMock.patchOnce('https://afe99802-9130-4320-a770-8d138b941e74.activities.api.proddev.d2l/6606/folders/13', editableEntity);
```
Comment (mdulat) [inline test/activities/CategoriesEntity.js:141]: This url seems strange to me `https://afe99802-9130-4320-a770-8d138b941e74.activities.api.proddev.d2l/6606/folders/13`. Does it actually exist? I.e. an `activities` api ending in `folders/<id>`?

Comment (ronan-f) [reply test/activities/CategoriesEntity.js:141]: Nope it doesn't exist...we would actually patch:

- `https://afe99802-9130-4320-a770-8d138b941e74.assignments.api.proddev.d2l/6606/folders/13`
- `https://afe99802-9130-4320-a770-8d138b941e74.quizzes.api.proddev.d2l/6606/quizzes/13`

but I guess I was trying to make it more generic so the next person who came along didn't think it was specific to a particular tool. 

I'll put up a new PR to improve this.

**Pull request merged**